pulp_ansible/app/viewsets.py

```diff
@@ -111,6 +167,77 @@ class CollectionViewset(NamedModelViewSet, mixins.RetrieveModelMixin, mixins.Lis
     queryset = Collection.objects.all()
     serializer_class = CollectionSerializer
     filterset_class = CollectionFilter
+    queryset_filtering_required_permission = "ansible.view_collection"
+
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list", "retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+            },
+            {
+                "action": "create",
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_perms:ansible.add_collection",
+            },
+            {
+                "action": "destroy",
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:ansible.delete_collection",
+                    "has_model_or_obj_perms:ansible.view_collection",
+                    "has_model_or_obj_perms:ansible.change_ansiblerepository",
```

Comment:
@mdellweg @gerrod3 does this api endpoint allow you to upload a collection without specifying a repo?

If so, we probably don't need to require repo permissions here.

Reply:
If you follow the example in pulp_file (https://github.com/pulp/pulp_file/blob/main/pulp_file/app/viewsets.py#L88-L92), then you won't be able to upload unless it is to a repository that you have perms on.

Reply:
I should also mention that this viewset isn't under the `/pulp/api/v3/` endpoints, it actually appears under `/ansible/collections/`. Personally I wouldn't add RBAC to these endpoints as the upload endpoint is deprecated and the list endpoint is kind of useless. If galaxy uses this viewset then maybe it would make sense to add a policy, otherwise I would leave it how it was.